pkg/provisioner/ironic/ironic.go

```diff
@@ -319,13 +341,73 @@ func (p *ironicProvisioner) changeNodeProvisionState(ironicNode *nodes.Node, opt
 	return result, nil
 }
 
+func getNICDetails(ifdata []introspection.InterfaceType) []metalkubev1alpha1.NIC {
+	nics := make([]metalkubev1alpha1.NIC, len(ifdata))
+	for i, intf := range ifdata {
+		nics[i] = metalkubev1alpha1.NIC{
+			Name:      intf.Name,
+			Model:     fmt.Sprintf("%s %s", intf.Vendor, intf.Product),
+			MAC:       intf.MACAddress,
+			Network:   "Pod Networking", // TODO(zaneb)
+			IP:        intf.IPV4Address,
+			SpeedGbps: 0, // TODO(zaneb)
+		}
+	}
+	return nics
+}
+
+func getStorageDetails(diskdata []introspection.RootDiskType) []metalkubev1alpha1.Storage {
+	storage := make([]metalkubev1alpha1.Storage, len(diskdata))
+	for i, disk := range diskdata {
+		storage[i] = metalkubev1alpha1.Storage{
+			Name:    disk.Name,
+			Type:    map[bool]string{true: "HDD", false: "SSD"}[disk.Rotational],
```

Comment:
Is it safe to say the type is either "HDD" or "SSD"? I know -- we live in the modern world but I assume there is some thought behind ironic's `rotational` bool.  I'm just speculating, but would the inspector pick up a USB thumb drive someone left in? If so, ironic would correctly call that non-rotational.  The code here would call that an SSD?

Reply:
Are you suggesting that thumb drives are not SSDs? ;)
`rotational` comes straight from udev so it's probably reasonably universal.